### PR TITLE
[libra_vm] Create a testing strategy for generating randomly invalid transactions

### PR DESCRIPTION
**Pull request by runtian-zhou**
## Motivation

Thanks to @sunshowers 's previous work, it is now relatively easy to describe a strategy to generate a random set of invalid transactions. This PR implements an easy strategy and the next step would be move the suite to fuzzer.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### CODE REVIEW

File: language/e2e-tests/src/account_universe/bad_transaction.rs [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+}
+
+/// Represents a insufficient balance transaction
+///
+#[derive(Arbitrary, Clone, Debug)]
+#[proptest(params = "(u64, u64)")]
+pub struct InsufficientBalanceGen {
+    sender: Index,
+    #[proptest(strategy = "params.0 ..= params.1")]
+    gas_unit_price: u64,
```
[the remark this answers is not in the document]
Comment (runtian-zhou) [reply language/e2e-tests/src/account_universe/bad_transaction.rs:76]: Should I add an assertion over there and provide a comment? The caller will provide in the argument so I would expect the caller to provide a non-zero lower bound.

[the remark this answers is not in the document]
Comment (tzakian) [reply language/e2e-tests/src/account_universe/bad_transaction.rs:76]: ah! No that's perfectly fine! I just misunderstood how these numbers were getting passed in is all :)

**Comment by runtian-zhou** [comment]
As a next step I'll refactor the tests here a bit so that the fuzzer can utilize the AccountUniverse to generate random transactions to test the LibraVM cc @tzakian @mimoo

**Comment by tzakian** [comment]
That sounds good to me. Another thing you might want to look at is possibly hooking this up with the "multi-random executor" (see `tests/execution_strategies.rs`); you can generate a block of transactions with proptest, and then the multi-executor will execute that block in multiple random segmentations and make sure the results are the same for each transaction.

**Pull request closed without merging**